packages/sync/src/offline/SyncOfflineMutation.ts

```diff
@@ -0,0 +1,75 @@
+import ApolloClient from "apollo-client";
+import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
+import { PersistentStore, PersistedData } from "../PersistentStore";
+import { OperationQueueEntry } from "../links/QueueLink";
+
+export class SyncOfflineMutation {
+    private apolloClient: ApolloClient<NormalizedCacheObject>;
+    private storage: PersistentStore<PersistedData>;
+    private storageKey: string;
+    private offlineData: OperationQueueEntry[] = [];
+
+    constructor(apolloClient: ApolloClient<NormalizedCacheObject>,
+                storage: PersistentStore<PersistedData>, storageKey: string) {
+        this.apolloClient = apolloClient;
+        this.storage = storage;
+        this.storageKey = storageKey;
+    }
+
+    public async init() {
+        const stored = await this.getOfflineData();
+        if (stored) {
+            this.offlineData = JSON.parse(stored.toString());
+        } else {
+            this.offlineData = [];
+        }
+    }
+
+    public sync = async () => {
+        // if there is no offline data  then just exit
+        if (!this.hasOfflineData()) { return; }
+
+        // return as promise, but in the end clear the storage
+        const uncommittedOfflineMutation: OperationQueueEntry[] = [];
+
+        await Promise.all(this.offlineData.map(async (item) => {
```

Comment:
It's critical to preserve order of operations. If edit mutations happen before create mutation then some requests will crash.